### PR TITLE
Create new GTA Version App

### PR DESCRIPTION
**Pull request by SergeyGhazaryan**

### CODE REVIEW

File: src/Components/index.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+
+import '../Styles/styles.css';
+
+export const GtaVersions = ({ imageLink, versionName, index }) => {
+  const GetLink = (index) => {
```
Comment (arayik-99) [inline src/Components/index.jsx]: You can remove this function and instead write the return value where it's being called.

File: src/VersionInformation/index.jsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React, { useMemo } from 'react';
+import { gtaModel } from '../constants/gtaModel';
+import { useParams } from 'react-router';
```
Comment (arayik-99) [inline src/VersionInformation/index.jsx]: Move this after react import

File: src/VersionInformation/index.jsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+const VersionInformation = () => {
+  let { id } = useParams();
+
+  const information = useMemo(() => {
+    if (gtaModel[id].information) return gtaModel[id].information;
+    return <div>No content</div>;
```
Comment (arayik-99) [inline src/VersionInformation/index.jsx]: This is wrong. Don't return it as component. You might use the value later.

File: src/App.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import VersionInformation from './VersionInformation';
+
+import './Styles/styles.css';
+
+const App = () => {
+  return (
+    <Router>
+      <Switch>
+        <Route path='/gta/:id' component={VersionInformation} />
+        <Route path='/' component={Layout} />
```
Comment (arayik-99) [inline src/App.js]: Move this upper.

File: src/Layout/index.jsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { gtaModel } from '../constants/gtaModel';
+import { GtaVersions } from '../Components';
+
+import '../Styles/styles.css';
```
Comment (arayik-99) [inline src/Layout/index.jsx]: Writing all of the styles in one css file is wrong. Each compoennt should have its own.

File: src/Components/index.jsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+
+import '../Styles/styles.css';
```
Comment (arayik-99) [inline src/Components/index.jsx]: Writing all of the styles in one css file is wrong. Each compoennt should have its own.

File: src/App.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
+import Layout from './Layout';
+import VersionInformation from './VersionInformation';
+
+import './Styles/styles.css';
```
Comment (arayik-99) [inline src/App.js]: Writing all of the styles in one css file is wrong. Each compoennt should have its own.

**Review by arayik-99 — CHANGES_REQUESTED** [review]

**Comment by arayik-99** [comment]
Rewrite the classNames to have kebab case (class-name), instead of camelCase (className). Use sass, scss, less or StyledComponents instead of regular css. Remake the folder structure. Remove the gtaModel from constants directory and create a directory (e..g. data/data.js). Rename gtaModel to something more meaningfull. Either make the directories lower case or upper case.

File: src/Layout/index.jsx [hunk 8]
```diff
@@ -1,20 +1,20 @@
 import React from 'react';
-import { gtaModel } from '../constants/gtaModel';
-import { GtaVersions } from '../Components';
+import { gtaData } from '../data/data';
```
Comment (arayik-99) [inline src/Layout/index.jsx]: you can rename the file in the folder data to index.js and just import data. Also rename the gtaData to just data.

File: src/App.js [hunk 9]
```diff
@@ -1,16 +1,14 @@
...
-import './Styles/styles.css';
+import Layout from './layout';
+import VersionInformation from './versionInformation';
 
 const App = () => {
   return (
     <Router>
       <Switch>
+        <Route exact path='/' component={Layout} />
         <Route path='/gta/:id' component={VersionInformation} />
```
Comment (arayik-99) [inline src/App.js]: place exact here as well. There might be some other routes like /gta/developers and /gta/engine

File: src/Layout/index.jsx [hunk 10]
```diff
@@ -1,20 +1,20 @@
...
+import { GtaVersions } from '../components/gtaVersions/index';
```
Comment (arayik-99) [inline src/Layout/index.jsx]: remove index. When importing files, if you are importing from the folder which has index.js/ts, you can just write the folder name.

File: src/VersionInformation/index.jsx [hunk 11]
```diff
@@ -1,13 +1,13 @@
-import React, { useMemo } from 'react';
-import { gtaModel } from '../constants/gtaModel';
+import { useMemo } from 'react';
 import { useParams } from 'react-router';
```
Comment (arayik-99) [inline src/VersionInformation/index.jsx]: Import from react-router-dom. Make sure everywhere there is something imported from react-router, import it from react-router-dom

**Review by arayik-99 — CHANGES_REQUESTED** [review]
Also don't forget to change the id to start from 1 for routing.

File: src/VersionInformation/index.jsx [hunk 12]
```diff
@@ -6,8 +6,9 @@ const VersionInformation = () => {
   let { id } = useParams();
 
   const information = useMemo(() => {
-    if (!data[id - 1].information) return;
-    return data[id - 1].information;
+    const necessaryData = data.find((d) => d.id == id);
```
Comment (arayik-99) [inline src/VersionInformation/index.jsx]: Put strict checking

**Review by arayik-99 — CHANGES_REQUESTED** [review]
Where is the data's id amends?

File: src/components/versionInformation/index.jsx [hunk 13]
```diff
@@ -0,0 +1,17 @@
+import { useMemo } from 'react';
+import { useParams } from 'react-router-dom';
+import { data } from '../../data';
+
+const VersionInformation = () => {
```
Comment (Karen-Avdalyan) [inline src/components/versionInformation/index.jsx]: change the name

File: src/components/gtaVersions/index.jsx [hunk 14]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+
+import './styles.scss';
+
+export const GtaVersions = ({ imageLink, versionName, id }) => {
```
Comment (Karen-Avdalyan) [inline src/components/gtaVersions/index.jsx]: it is a gta version not versions

File: src/Layout/index.jsx [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+import { data } from '../data';
+import { GtaVersions } from '../components/gtaVersions';
+
+import './styles.scss';
+
+const Layout = () => {
+  return (
+    <>
+      <div className='header'>GTA Versions</div>
+      <div className='gta-version'>
```
Comment (Karen-Avdalyan) [inline src/Layout/index.jsx]: change class name

File: src/Layout/index.jsx [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+        <div className='version-component'>
```
Comment (Karen-Avdalyan) [inline src/Layout/index.jsx]: change class name

File: src/components/gtaVersions/index.jsx [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+  return (
+    <div className='version-container'>
+      <NavLink to={`/gta/${id}`} className='nav-link'>
+        <h1 className='image-text'>{versionName}</h1>
```
Comment (Karen-Avdalyan) [inline src/components/gtaVersions/index.jsx]: change class name

File: src/components/versionInformation/index.jsx [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+  let { id } = useParams();
```
Comment (Karen-Avdalyan) [inline src/components/versionInformation/index.jsx]: use const

File: src/components/versionInformation/index.jsx [hunk 19]
```diff
@@ -0,0 +1,17 @@
...
+
+  const information = useMemo(() => {
+    const necessaryData = data.find((d) => d.id == id);
```
Comment (Karen-Avdalyan) [inline src/components/versionInformation/index.jsx]: change the name

File: src/components/versionInformation/index.jsx [hunk 20]
```diff
@@ -0,0 +1,17 @@
+import { useMemo } from 'react';
+import { useParams } from 'react-router-dom';
+import { data } from '../../data';
+
+const VersionInformation = () => {
+  let { id } = useParams();
+
+  const information = useMemo(() => {
```
Comment (Karen-Avdalyan) [inline src/components/versionInformation/index.jsx]: No need for useMemo

**Review by Karen-Avdalyan — CHANGES_REQUESTED** [review]

**Pull request merged**